### PR TITLE
[SPARK-6724][WIP][MLLIB]Model import/export for FPGrowth

### PR DESCRIPTION
**Pull request by FlytxtRnD**

### CODE REVIEW

**Comment by jkbradley** [comment]
There are some merge conflicts.  Could you please fix them, and then I'll try to test this out myself?

**Comment by FlytxtRnD** [comment]
@jkbradley  PR is  updated to solve the merge issues. Please have a look at the issue.

**Comment by jkbradley** [comment]
@FlytxtRnD I should have realized this before, but I think this may be fairly difficult to do.  ML save/load relies on DataFrame serialization, so Item needs to be a SQL DataType.  I think you will need to use Scala reflection to see if the Item type is a DataType, and only save if so.  Are you familiar with Scala reflection?

**Comment by FlytxtRnD** [comment]
@jkbradley We  will give it a try and let you know if we are not able to move forward.

**Comment by FlytxtRnD** [comment]
@jkbradley Plz have a look at the code.  I guess the save function is ok. Now I am facing issues with Loader. Can you please help me with passing the type parameter for freqItemSet class here ` object FPGrowthModel extends Loader[FPGrowthModel[FreqItemset]]` ?

File: mllib/src/main/scala/org/apache/spark/mllib/fpm/FPGrowth.scala [hunk 1]
```diff
@@ -52,8 +62,74 @@ class FPGrowthModel[Item: ClassTag](val freqItemsets: RDD[FreqItemset[Item]]) ex
...
+
+      def save[Item: ClassTag](sc: SparkContext, model: FPGrowthModel[Item], path: String): Unit = {
+        val sqlContext = new SQLContext(sc)
+
+        val metadata = compact(render(
+          ("class" -> thisClassName) ~ ("version" -> thisFormatVersion) ~
+            ("frequentItemsCount" -> model.freqItemsets.count())))
+        sc.parallelize(Seq(metadata), 1).saveAsTextFile(Loader.metadataPath(path))
+        val sqlType = model.freqItemsets.first().items(0) match {
+          case _: java.lang.String => StringType
```
Comment (feynmanliang) [inline mllib/src/main/scala/org/apache/spark/mllib/fpm/FPGrowth.scala]: This code assumes that items are a primitive type, but items are currently arbitrary. We should discuss on the JIRA whether it's acceptable to restrict items to be primitives / types supported in `spark.sql.types`

File: mllib/src/test/scala/org/apache/spark/mllib/fpm/FPGrowthSuite.scala [hunk 2]
```diff
@@ -274,4 +276,30 @@ class FPGrowthSuite extends SparkFunSuite with MLlibTestSparkContext {
      */
     assert(model1.freqItemsets.count() === 65)
   }
+
```
Comment (feynmanliang) [inline mllib/src/test/scala/org/apache/spark/mllib/fpm/FPGrowthSuite.scala]: Once you're done working, can you uncomment these tests so Jenkins runs them?

**Comment by jkbradley** [comment]
@FlytxtRnD I'm sorry I wasn't able to devote time to this yet, but thank you for your efforts.  I'll monitor https://github.com/apache/spark/pull/9267 now and will appreciate your help in testing that PR if you still have a need for this feature.

Could you please close this issue now that it has been replaced with the other PR?  Thank you!

**Comment by FlytxtRnD** [comment]
@jkbradley Thank you for the review. I will close this issue and will surely help in testing the other PR. Also it would be great if you could take a look at #6880

**Pull request closed without merging**